src/main/java/nl/workingtalent/wtacademy/review/ReviewController.java

```diff
@@ -1,11 +1,154 @@
+
+    @GetMapping("review/book/{bookId}")
+    public ResponseDto getAllReviews(@PathVariable("bookId") long bookId, HttpServletRequest request, @RequestBody int pageNr) {
+        Optional<Book> bookOptional = bookService.getBookById(bookId);
+        if (bookOptional.isEmpty()) {
+            return new ResponseDto(false, null, null, "Book not found.");
+        }
+        Book book = bookOptional.get();
+
+        User requestUser = (User) request.getAttribute("WT_USER");
+        if (requestUser == null) {
+            return new ResponseDto(false, null, null, "No user found.");
+        }
+
+        Pageable pageable = PageRequest.of(pageNr, pageSize);
+        Page<Review> reviews = service.getAllReviewsForBook(bookId, pageable);
+        Stream<ReadReviewDto> dtos = reviews.stream().map((review) -> {
+            return new ReadReviewDto(review);
+        });
+
+        return new ResponseDto(true, dtos, null, "Reviews found.");
+    }
+
+    @PostMapping("review/create")
+    public ResponseDto createReview(@RequestBody CreateReviewDto dto, HttpServletRequest request) {
+        User requestUser = (User) request.getAttribute("WT_USER");
+        if (requestUser == null) {
+            return new ResponseDto(false, null, null, "No user found.");
+        }
+
+        Optional<Book> bookOptional = bookService.getBookById(dto.getBookId());
+        if (bookOptional.isEmpty()) {
+            return new ResponseDto(false, null, null, "Book not found.");
+        }
+        Book book = bookOptional.get();
+
+//		Check if user has borrowed and returned the book
+        boolean hasReturnedBook = requestUser.getLoans().stream()
+                .anyMatch(loan -> loan.getBookCopy().getBook().getId() == book.getId() && !loan.isActive());
+
+        if (!hasReturnedBook) {
+            return new ResponseDto(false, null, null, "You have not returned the book yet.");
+        }
+
+        Review review = new Review();
+//		Check rating
+        if (dto.getRating() < 1 || dto.getRating() > 5) {
+            return new ResponseDto(false, null, null, "Rating must be between 1 and 5.");
+        } else {
+            review.setRating(dto.getRating());
+        }
+
+//		Check entered date
+        if (dto.getDate() != null) {
+            review.setDate(dto.getDate());
+        } else {
+            review.setDate(LocalDate.now());
+        }
+
+        if (dto.getReviewText() != null) {
+            review.setReviewText(dto.getReviewText());
+        }
+
+        review.setBook(book);
+        review.setUser(requestUser);
+        service.addReview(review);
+
+        return new ResponseDto(true, null, null, "Review created.");
+    }
+
+    @PutMapping("review/update")
+    public ResponseDto updateReview(UpdateReviewDto dto, HttpServletRequest request) {
+        User requestUser = (User) request.getAttribute("WT_USER");
+        if (requestUser == null) {
+            return new ResponseDto(false, null, null, "No user found.");
+        }
+
```

Comment:
Update werkt bij mij niet, snap niet waarom maar als ik debug zijn alle waardes van UpdateReviewDto null of 0. Ook al zijn ze wel ingevuld.